temporal_sqlalchemy/clock.py

```diff
@@ -131,24 +131,30 @@ def make_temporal(cls: nine.Type[Clocked]):
 
         clock_properties = dict(
             __tablename__=clock_table_name,
-            # todo support different shape PKs
-            entity_id=sa.Column(sa.ForeignKey(cls.id), primary_key=True),
+
+            entity_id=sa.Column(sa.ForeignKey(cls.id), nullable=False),
             entity=orm.relationship(
                 cls, backref=orm.backref("clock", lazy='dynamic')),
         )
 
+        tick_entity_constraint_name = truncate_identifier(
+            '%s_tick_entity_id_key' % clock_table_name
+        )
+        table_args = [
+            sa.UniqueConstraint('tick', 'entity_id',
+                                name=tick_entity_constraint_name)
+        ]
+
         if activity_cls is not None:
             backref_name = '%s_clock' % entity_table_name
             clock_properties['activity_id'] = sa.Column(
                 sa.ForeignKey(activity_cls.id), nullable=False)
             clock_properties['activity'] = orm.relationship(
                 activity_cls, backref=backref_name)
-            clock_properties['__table_args__'] = (
-                sa.UniqueConstraint('entity_id', 'activity_id'),
-                {'schema': schema}
-            )
-        else:
-            clock_properties['__table_args__'] = {'schema': schema}
+            table_args.append(sa.UniqueConstraint('entity_id', 'activity_id'))
```

Comment:
Yes it creates a `btree` index:

```
\d+ trg_history.guac_prior_auth_correspondence_events_clock
...
Indexes:
    "guac_prior_auth_correspondence_events_clock_pkey" PRIMARY KEY, btree (id)
    "guac_prior_auth_correspondence_events_clock_tick_entity_id_key" UNIQUE CONSTRAINT, btree (tick, entity_id)
```

And I've verified it will use an index scan:

```
dev_app_db=# EXPLAIN SELECT * FROM trg_history.guac_prior_auth_correspondence_events_clock ORDER BY tick, entity_id;
                                                                              QUERY PLAN
----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Index Scan using guac_prior_auth_correspondence_events_clock_tick_entity_id_key on guac_prior_auth_correspondence_events_clock  (cost=0.15..57.95 rows=920 width=60)
```